feat: add auto-switch network hook for hardware and QR wallets in web…

limichange opened the pull request:
… DApp mode

- Introduced `useAutoSwitchNetworkForHwQrWallet` hook to automatically switch to "All Networks" when a hardware or QR wallet is detected.
- Integrated the new hook into `AccountSelectorEffectsCmp` for seamless functionality.
- Ensured the feature is only active on the Web DApp platform and avoids redundant updates.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * In Web DApp mode, hardware and QR wallet users will now be silently and automatically switched to the "All Networks" view when the wallet changes and such a switch is applicable. This removes the need for manual network selection in those scenarios and prevents redundant network updates, improving continuity when switching external wallets.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: add auto-switch network hook for hardware and QR wallets in web DApp mode

- Introduced `useAutoSwitchNetworkForHwQrWallet` hook to automatically switch to "All Networks" when a hardware or QR wallet is detected.
- Integrated the new hook into `AccountSelectorEffectsCmp` for seamless functionality.
- Ensured the feature is only active on the Web DApp platform and avoids redundant updates.

limichange committed:
Merge branch 'x' into OK-44822

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/c63a0cea-2b16-48e4-9dd9-8ca45178226c/license?refs=f98d01ea965ffa76cebaf7643e4e6479ee45dabe&source=prChecksComment) |
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/c63a0cea-2b16-48e4-9dd9-8ca45178226c?refs=f98d01ea965ffa76cebaf7643e4e6479ee45dabe&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix reviewed (commented):
那 hw 的钱包无法用单链了么，感觉不合理吧

limichange committed:
refactor: enhance auto-switch network logic for hardware and QR wallets

- Updated `useAutoSwitchNetworkForHwQrWallet` to track previous wallet ID and ensure auto-switching only occurs when transitioning from an external wallet to a hardware or QR wallet.
- Improved documentation to clarify the conditions under which the auto-switch feature activates.
- Maintained functionality to skip redundant updates when already on "All Networks".

limichange committed:
Merge branch 'x' into OK-44822
